### DISCUSSION TITLE
REM Intellibrite lights

### DISCUSSION THREAD

**Discussion by orrious** [tagyoureit/nodejs-poolController#499] [REM]
My transformer is switched as well as each Intellibrite light; I have three.  I switch the transformer so it is not running unless the lights are active.  I switch each light individually so that I can run each independently or in unison.  It's a nice effect when you stagger the lights 5 seconds between each of the three.  The transform remains on for 30 seconds after the last light is turned off.  I use solid state relays for all switching; intellitbrite lights are reliably able to be switched on / off at 2hz (0.5 seconds per cycle) for mode change.

Does REM currently have the logic to control each of these components discreetly and create scenes as mentioned above?

**Comment by tagyoureit** [comment]
Yup, you sure can.  Create three circuits and set them to `Intellibrite`.  Then create a Light Group with those three circuits.  This will take care of everything assuming your transformer relay is on.  

You could also add the extra relay for the transformer and put it on its own circuit.  The ability to switch on the transformer, wait 30s, and then turn on the lights isn't something we support natively.  Have you actually measured how much electricity is consumed when the transformer is not under load?  It may not be very much.

I don't recall at the moment if you are using node-rem, or another platform, but you could kick off some logic from an external platform.  I'll give it some more thought, too, but there isn't currently any logic in the platform that would support this.

**Reply by orrious** [reply]
The 30 second delay is from ON to OFF.  This is to avoid turning off the transformer during mode changes.  The whole intent is to only run the transformer when any light is on.  When the last light goes off, initiate a 30 second count down timer.  If the timer reaches 0, turn the power off.  This prevents cycling power to the transformer too much.

**Comment by orrious** [comment]
Are you able to create "scenes" with the lights?  Ex: Scene Evening: light 1: sunset, delay 2 seconds, light 2: sunset, delay 2, light 3: sunset.  I already do this with my own setup, I'm inquiring about feature parity before moving over.  Ironically I remember long conversations with you and @rstrouse about njs-PC being in complete control of the pool and not relying on outside automation.  It took 2 years, but I've come to see your view and am striving towards that goal.

**Reply by tagyoureit** [reply]
You can sync all lights with a light show:
![2022-06-23_12-13-03](https://user-images.githubusercontent.com/1646840/175379513-07df3428-72e6-47f2-895c-e3f59203eec4.png)

You can set individual colors/delays and have them sync/set/swim:
![2022-06-23_12-16-10](https://user-images.githubusercontent.com/1646840/175379726-59a3c400-ea33-4d75-b7f9-681864a7d607.png)

You can set individual lights to a theme:
![2022-06-23_12-14-00](https://user-images.githubusercontent.com/1646840/175379755-758b6109-1996-4442-946d-98139af2442d.png)

Pentair OCP doesn't allow delays between different themes on different lights (unless IC does, but not sure...).  You could easily make this happen manually or by scripting the API's to turn on each light 2s after the previous.  We don't have that native in the app, tho.

**Reply by orrious** [reply]
How does REM specifically handle Intellibrite lights?  I've heard that OCP is busy during mode changes and isn't able to do it concurrently.  Is REM able to concurrently mode switch multiple lights at the same time, but not synchronized to the same mode?

**Reply by tagyoureit** [reply]
If you use a Light Group then the controller will be controlling multiple IB at once and it will go into the 'busy' mode.  This is because it needs to reset all the lights and then possible set the timing between them.  But if you set the themes on individual circuits/intellibrite lights, then they do their individual sequencing and you can set each one individually at any time.

**Reply by tagyoureit** [reply]
You know you could setup a Nixie controller, not connected to any equipment at all, and test this all out... It will show you how everything works without having to dismantle what you have physically connected now.

**Reply by rstrouse** [reply]
While REM does a better job of it because it does not tie up the processor while switching.  The limitation that the OCP and REM face is that the light protocol borders on idiotic.   The process we use is to batch up a series of states with timing and send it off to the REM processor.   Then the REM process frees up the processor between state flips.  

Sadly this switchy flippy thingy that pool lighting from all manufacturers does is so contrary to usable that it always falls short of great light shows.  One of these days, the dude I share my chair with will get around to creating a DMX512 or WS2812 board to replace the light engine.  Then we will be cooking with gas.

Imagine building pool light shows like I can for my landscape lighting.
![image](https://user-images.githubusercontent.com/47839015/175609181-f2eab9e7-1d32-442b-8558-e44a5e7988bd.png)

**Reply by orrious** [reply]
@tagyoureit That's my next step, but getting a spare Rpi is like pulling hens teeth right now.

@rstrouse I don't think it's a limitation of REM per se, but a limitation of the actual intellibrite / globrite light itself and how you interact with it.  I 100% agree on your desire to use DMX controlled lights, but I've yet to be able to find any.  2 of my 3 globrite lights are being replaced under warranty.  I'm hoping I can get them to leave the old ones so I can crack them open.

The way I implemented it in my current iteration is to tell my controller which mode I want it on and it does all the timing, etc for me.  I didn't realize the actual timing was sent from OCP / njs-PC

Can you share which landscape lights you're using

**Reply by rstrouse** [reply]
I use a Nicolaude Stick DE3 with two DMX512 universes.  Then I cobbled together some DMX decoders and hooked them up to these lights.
https://www.ecolocityled.com/product/rgbw-underwater-led/under-water-led-light

I also built some RGBW LED hanging path lights that also tie into the fixture map.

The picture really doesn't do it justice but there is also a sound to light processor that does full on music decoding to each fixture.
![image](https://user-images.githubusercontent.com/47839015/175614331-0e437716-f678-4aa7-b2e4-10419699912c.png)

**Reply by orrious** [reply]
We should start a thread somewhere for landscape lighting, sound integration, ignition systems, etc.  After I get migrated to Nixie 100% I'm turning my focus back to this.

**Reply by tagyoureit** [reply]
> @tagyoureit That's my next step, but getting a spare Rpi is like pulling hens teeth right now.

You can install the software on a PC, Mac, or Linux.  You don't even need to connect to REM.  If REM detects it isn't on a compatible platform it will just use mock values instead of trying to write to the correct bus.

**Reply by mguinness** [reply]
Hayward have [Universal ColorLogic](https://hayward-pool-assets.com/assets/documents/pools/pdf/literature/ColorLogic_LED_Lighting_Family_Trifold_Brochure_LITLEDFAM19.pdf) lights which have the following features, but unclear if OmniDirect mode uses the DMX protocol.

- Instant navigation to desired colors or light shows (no more cycling through every option)
- Dimming capabilities (20% increments)
- Speed adjustment for light shows
- Access to 10 additional colors for increased customization opportunities (special occasions, sports teams, school colors and more)

**Reply by rstrouse** [reply]
I am pretty sure it is not DMX.  I looked at this AQL-COLOR-MODHV modem they use on the power line side and think it is just a variant on Powerline communication.  Heck it might actually be a powerline modem that simply switches the modes.  Then there is another LKBUN1000 coupler component that does the last mile for the comms.  In the gen4 lights they put this component in the light.

Yet another half measure for something so expensive.  What is nuts is they want 500 bones for what appears to be a powerline modem.  Then another 100 bones for the receiver.

**Reply by mguinness** [reply]
That's lame then.  Have you seen the [Mini Brio RX3 light](https://na.ccei-pool.com/us/dmx-lighting-solution/370-mini-brio-liner.html) by CCEI?

**Reply by rstrouse** [reply]
Now they are heading in the right direction.  I have seen a similar solution for niche less designs where they appear to put the entire fixture on the surface of the pool.   You can buy the 1.5 inch CCIE but the 10 inch looks a bit scarce at this point.  I wonder if it is capable of sharing a universe.

https://www.poolsupplyunlimited.com/pool/ccei-pf11r010-mini-brio-rx3-light-rgbw-dmx-solution/179414p1
